contrib/generic_proxy/filters/network/source/stats.cc

```diff
@@ -33,29 +35,13 @@ Stats::StatName CodeOrFlags::statNameFromCode(uint32_t code) const {
 }
 
 Stats::StatName CodeOrFlags::statNameFromFlag(StreamInfo::ResponseFlag flag) const {
-  const auto iter = flag_stat_names_.find(flag);
-  if (iter != flag_stat_names_.end()) {
-    return iter->second;
+  ASSERT(flag.value() < flag_stat_names_.size());
```

Comment:
Hmmm, this if check is kept because previous code.

But I believe the unknown code path can never be reached because the responseFlagsVec() will contains all valid response flags.

I will update it.

Reply:
I agree with your view, but just want to double check my understanding:
The input to this function will always be created by some internal Envoy component, correct? otherwise (let's say the response-flag is an int coming from a downstream/upstream and is passed to this function) then we may need to handle the unknown-flag case.

Reply:
> The input to this function will always be created by some internal Envoy component, correct?

Yeah. It not allowed to creat ResponseFlag from int directly because of the private constructor. It could only be created from limited enum and the `registerCustomFlag` method. All these flags must be contained by the `StreamInfo::ResponseFlagUtil:: responseFlagsVec()`.

Reply:
Ack, thanks!